### PR TITLE
V7.1.0

### PR DESCRIPTION
**Pull request by rmarinleal**
* New version with orders
* Updating dependencies and vendor versions
* Module internationalization to English, French, Italian and Portuguese
* Module standardization code.
* Module Backoffice redesign
* Module advanced options behavior
* Module compatibility with prestashops versions 1.5, 1.6 and 1.7 in latests versions.
* New logs managing
* Increase testing

### CODE REVIEW

File: controllers/front/AbstractController.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+    /**
+     * EXCEPTION RESPONSES
+     */
+    const CC_ERR_MSG = 'Unable to block resource';
+    const CC_NO_MERCHANT_ORDERID = 'Merchant Order Id (cart_id) not found';
+    const CC_NO_CONFIG = 'Unable to load module configuration';
+    const CC_MALFORMED = 'Bad request, module may not be enabled';
+
+    const GMO_ERR_MSG = 'Unable to find merchant Order';
```
Comment (romeritoCL) [inline controllers/front/AbstractController.php]: Please explain me this constants I do not understand them

File: sql/install.sql [hunk 2]
```diff
@@ -1,5 +1,18 @@
-CREATE TABLE `PREFIX_pmt_cart_process` (
+CREATE TABLE IF NOT EXISTS `PREFIX_pmt_cart_process` (
   `id` INT NOT NULL ,
   `timestamp` INT NOT NULL ,
   PRIMARY KEY (`id`)
   ) ENGINE = InnoDB;
+
+CREATE TABLE IF NOT EXISTS `PREFIX_pmt_order` (
```
Comment (romeritoCL) [inline sql/install.sql]: A partir de que versión de MySQL se puede usar el IF NOT EXISTS?

ten en cuenta que puede ser un mysql muy antiguo

File: sql/uninstall.sql [hunk 3]
```diff
@@ -1 +1,2 @@
-DROP TABLE `PREFIX_pmt_cart_process`;
+DROP TABLE IF EXISTS `PREFIX_pmt_cart_process`;
```
Comment (romeritoCL) [inline sql/uninstall.sql]: Same

File: test/advanced/PaylaterPs17AdvancedTest.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        // Check min amount simulator
+        // no longer available with .env file
+        // $this->loginToBackOffice();
+        // $this->getPaylaterBackOffice();
+        // $this->findById('pmt_display_min_amount')->clear()->sendKeys(500);
+        // $this->findById('module_form_submit_btn')->click();
+        //
+        // $this->goToProduct(false);
+        // $html = $this->webDriver->getPageSource();
+        // $this->assertNotContains('PmtSimulator', $html);
```
Comment (romeritoCL) [inline test/advanced/PaylaterPs17AdvancedTest.php]: Codigo comentado

File: test/advanced/PaylaterPs16AdvancedTest.php [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        // $this->findById('pmt_display_min_amount')->clear()->sendKeys(500);
+        // $this->findById('module_form_submit_btn')->click();
+        //
+        // $this->goToProduct(false);
+        // $html = $this->webDriver->getPageSource();
+        // $this->assertNotContains('PmtSimulator', $html);
+        //
+        // Hide simulator
+        // $this->getPaylaterBackOffice();
+        // $this->findById('pmt_display_min_amount')->clear()->sendKeys(1);
```
Comment (romeritoCL) [inline test/advanced/PaylaterPs16AdvancedTest.php]: Código comentado

File: test/advanced/PaylaterPs15AdvancedTest.php [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        // $html = $this->webDriver->getPageSource();
+        // $this->assertNotContains('PmtSimulator', $html);
+        //
+        // //Hide simulator
+        // $this->getPaylaterBackOffice();
+        // $this->findById('pmt_display_min_amount')->clear()->sendKeys(1);
+        // $this->findById('pmt_simulator_is_enabled_off')->click();
+        // $this->findById('module_form')->submit();
+        //
+        // $this->goToProduct(false);
```
Comment (romeritoCL) [inline test/advanced/PaylaterPs15AdvancedTest.php]: Código comentado

File: controllers/front/AbstractController.php (hunk 1 above)
Comment (romeritoCL) [reply controllers/front/AbstractController.php]: Please export them to module utils and use them from there

File: controllers/front/AbstractController.php [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+     * @param bool   $error
+     * @param string $url
+     * @param array  $parameters
+     */
+    public function redirect($error = true, $url = '', $parameters = array())
+    {
+        if ($error) {
+            Tools::redirect($url);
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (romeritoCL) [reply controllers/front/AbstractController.php]: maybe even think of delete it 🔢

File: controllers/front/AbstractController.php (hunk 1 above)
Comment (rmarinleal) [reply controllers/front/AbstractController.php]: Deleted constans

File: controllers/front/AbstractController.php [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+//        $response = $this->getResponse();
+//
+//        $output = array();
+//        if (!empty($this->errorMessage)) {
+//            $output['errorMessage'] = $this->errorMessage;
+//        }
+//        if (!empty($this->errorDetail)) {
+//            $output['errorDetail'] = $this->errorDetail;
+//        }
+//        if (count($extraOutput)) {
```
[the remark this answers is not in the document]
Comment (rmarinleal) [reply controllers/front/AbstractController.php]: Removed

File: controllers/front/AbstractController.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (rmarinleal) [reply controllers/front/AbstractController.php]: Refactor function

File: controllers/front/api.php [hunk 9]
```diff
@@ -1,11 +1,4 @@
 <?php
-/**
- * This file is part of the official Paylater module for PrestaShop.
- *
```
[the remark this answers is not in the document]
Comment (rmarinleal) [reply controllers/front/api.php]: resolved

File: test/advanced/PaylaterPs17AdvancedTest.php (hunk 4 above)
Comment (rmarinleal) [reply test/advanced/PaylaterPs17AdvancedTest.php]: Removed

File: test/advanced/PaylaterPs16AdvancedTest.php (hunk 5 above)
Comment (rmarinleal) [reply test/advanced/PaylaterPs16AdvancedTest.php]: Removed

File: test/advanced/PaylaterPs15AdvancedTest.php (hunk 6 above)
Comment (rmarinleal) [reply test/advanced/PaylaterPs15AdvancedTest.php]: Removed

File: sql/install.sql (hunk 2 above)
Comment (rmarinleal) [reply sql/install.sql]: PHP 5.3 released at 30 June 2009
The first release (not beta) of mysql was Version 5.1 at November 2008 (next was 5.5 in 2010) so the function must exists in 5.q version...checking

Comment (rmarinleal) [reply sql/install.sql]: Is supported.
http://ftp.nchu.edu.tw/MySQL/doc/refman/5.1/en/create-table.html

File: sql/uninstall.sql (hunk 3 above)
Comment (rmarinleal) [reply sql/uninstall.sql]: Supported

File: controllers/front/AbstractController.php [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+    {
+        $httpStatusCodes = array(
+            200 => "OK",
+            201 => "Created",
+            202 => "Accepted",
+            400 => "Bad Request",
+            401 => "Unauthorized",
+            402 => "Payment Required",
+            403 => "Forbidden",
+            404 => "Not Found",
```
[the remark this answers is not in the document]
Comment (rmarinleal) [reply controllers/front/AbstractController.php]: Removed unused method

File: controllers/front/AbstractController.php [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+     * @param array $data
+     */
+    public function saveLog($data = array())
+    {
+        try {
+            $data = array_merge($data, array(
+                'timestamp' => time(),
+                'date' => date("Y-m-d H:i:s"),
+            ));
+
```
[the remark this answers is not in the document]
Comment (rmarinleal) [reply controllers/front/AbstractController.php]: Refactor to use LogEntry from module-utils

File: controllers/front/log.php [hunk 12]
```diff
@@ -0,0 +1,69 @@
+<?php
```
[the remark this answers is not in the document]
Comment (rmarinleal) [reply controllers/front/log.php]: Resolved

File: package.json [hunk 13]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "paylater",
-  "version": "1.0.0",
+  "version": "7.0.5",
```
[the remark this answers is not in the document]
Comment (rmarinleal) [reply package.json]: Resolved

File: paylater.php [hunk 14]
```diff
@@ -31,6 +30,11 @@ class Paylater extends PaymentModule
      */
     public $bootstrap = true;
 
+    /**
+     * @var _errors
+     */
+    public $_errors = array();
```
[the remark this answers is not in the document]
Comment (rmarinleal) [reply paylater.php]: Renamed

File: paylater.php [hunk 15]
```diff
@@ -53,20 +58,33 @@ public function __construct()
...
         }
 
         parent::__construct();
     }
 
     /**
-     * Configure the database variables for paga+tarde payment method.
-     *
+     * Configure the variables for paga+tarde payment method.
+     *º
```
[the remark this answers is not in the document]
Comment (rmarinleal) [reply paylater.php]: Resolved

**Pull request merged**